packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,334 @@
+        field-name="progress"
+        placeholder="Not started"
+    >
+        Progress
+    <my-app-progress-column>`
+</nimble-table>
+```
+
+**Define and register column custom element**
+
+```ts
+export class MyAppProgressColumn extends TableColumnTextBase {
+    public constructor() {
+        super({
+            cellRecordFieldNames: ['value'],
+            cellViewTag: tableColumnNumericTextCellViewTag,
+            groupHeaderViewTag: tableColumnNumericTextGroupHeaderTag
+        });
+        this.columnInternals.sortOperation = TableColumnSortOperation.basic;
+    }
+}
+
+const myAppProgressColumn = MyAppProgressColumn.compose({
+    baseName: 'table-column-progress',
+    template,
+    styles
+});
+
+DesignSystem.getOrCreate().withPrefix('my-app').register(myAppProgressColumn());
+export const myAppProgressColumnTag = DesignSystem.tagFor(MyAppProgressColumn);
+```
+
+**Specify formatting of cells and group headers in their custom elements**
+
+```ts
+export class MyAppProgressColumnCellView extends TableColumnTextCellViewBase<
+    TableColumnNumericTextCellRecord,
+    TableColumnNumericTextColumnConfig
+> {
+    public override get text(): string {
+        return `${100 * this.cellRecord.value}%`;
+    }
+
+    public override get placeholder(): string {
+        return this.columnConfig.placeholder;
+    }
+
+    public override get shouldUsePlaceholder(): boolean {
+        return typeof this.cellRecord.value !== 'number';
+    }
+}
+
+export class MyAppProgressColumnGroupHeaderView extends TableColumnTextGroupHeaderViewBase<
+    TableNumberFieldValue,
+    TableColumnNumericTextColumnConfig
+> {
+    public override get text(): string {
+        return `${100 * this.groupHeaderValue}%`;
+    }
+
+    public override get placeholder(): string {
+        return this.columnConfig.placeholder;
+    }
+
+    public override get shouldUsePlaceholder(): boolean {
+        return typeof this.groupHeaderValue !== 'number';
+    }
+}
+
+// Not shown: registering the above two custom elements
+```
+
+This is prototyped in the [formatted-text-column branch](https://github.com/ni/nimble/compare/main...users/jattas/formatted-text-column?expand=1).
+
+### Nimble formatted text columns
+
+Nimble will provide several columns that derive from the above base classes and provide higher level text formatting APIs for specific data types.
+
+#### Column naming
+
+Columns which render as text will use the following naming scheme: `nimble-table-column[-type]-text`.
+
+-   All columns start with `nimble-table-column-`.
+-   The `-type` section distinguishes between columns that render as text but require different configuration (e.g. `-numeric` or `-date`). We omit it for `nimble-table-column-text` since it seems redundant and would be a low-value breaking change.
+-   Adding `-text` allows for future column types that render the same data in different ways with different configuration. e.g. `numeric-progress`, `numeric-input`.
```

Comment:
```suggestion
#### Column naming

Follows the [/packages/nimble-components/src/table/specs
/table-columns-hld.md#column-type-philosophy](Column Type Philosophy) where:
- category: `table-column`
- presentation: `text`
- variants: Different variants are allowed for configurations that vary significantly / don't make sense to add to `table-column-text`, ie for a `table-column-numeric-text` or `table-column-date-text`
```

Reply:
Thank you, I forgot to update this file after moving this guidance around.